File: develop/server/project/game/src/main/java/com/home/game/logic/GCharacterUseLogic.java
package com.home.game.logic;

import com.home.base.data.role.GCharacterUseData;
import com.home.commonBase.data.role.MUnitUseData;
import com.home.commonGame.logic.unit.CharacterUseLogic;

/** 游戏角色使用逻辑 */
public class GCharacterUseLogic extends CharacterUseLogic
{
	private GCharacterUseData _data;
	
	@Override
	protected void setData(MUnitUseData data)
	{
		super.setData(data);
		
		_data=(GCharacterUseData)data;
	}
	
	/** 游戏数据 */
	public GCharacterUseData getGData()
	{
		return _data;
	}
	
}

File: develop/client/game/Assets/src/game/logic/GCharacterUseLogic.cs
using System;
using ShineEngine;

/// <summary>
/// 游戏角色使用逻辑
/// </summary>
public class GCharacterUseLogic:CharacterUseLogic
{
	private GCharacterUseData _data;

	protected override void setData(MUnitUseData data)
	{
		base.setData(data);

		_data=(GCharacterUseData)data;
	}

	/** 游戏数据 */
	public GCharacterUseData getGData()
	{
		return _data;
	}
}